extensions/mo_rrtmgp_clr_all_sky.F90

```diff
@@ -85,6 +85,8 @@ function rte_lw(k_dist, gas_concs, p_lay, t_lay, p_lev,    &
     ngpt  = k_dist%get_ngpt()
     nband = k_dist%get_nband()
 
+    !$acc update if_present host(p_lay(1,1), p_lay(1,nlay:nlay))
+    !$omp target update from(p_lay(1,1), p_lay(1,nlay:nlay))
```

Comment:
Why not just the `update` the entire matrix? Is it too large? Principle of least surprise -- that way you don't end up with a partially update matrix on the host.

Reply:
It is way too large to update the full matrix. Maybe there's a better way to compute top_at_1, or get it as an input to top-level radiation routines, @RobertPincus what do you think?

Reply:
While it gets the job done, my main concern is that this could surprise someone unfamiliar with the code. The GPU has the updated matrix, but the CPU only has two matrix elements updated. I supposed it would be OK if you add a comment/warning in the code explaining why this is done in this particular way.

Reply:
I can change to a bit different approach then. I can place the comparison into the `acc kernels` construct and copyout the result. Does OpenMP provide something similar to `acc kernels`, or just an offload parallel region with just 1 thread?

Reply:
The closet thing to `acc kernels` is `omp declare target`, but this is mostly used for calling small user-defined functions from inside a target region. If you just need one thread to do it on the GPU, you can just do `$omp start target` and `$omp end target`. Here is an example:
https://github.com/earth-system-radiation/rte-rrtmgp/blob/main/rrtmgp/mo_gas_optics_rrtmgp.F90#L738-L743

Reply:
Great, that should do. Thanks!

Reply:
@alexeedm Will you update this?

Reply:
resolved